spring-cloud-kubernetes-ribbon/src/main/java/org/springframework/cloud/kubernetes/ribbon/KubernetesServerList.java

```diff
@@ -53,60 +57,117 @@
 
 	private String portName;
 
+	private KubernetesRibbonProperties properties;
+
 	public KubernetesServerList(KubernetesClient client) {
 		this.client = client;
 	}
 
+	public KubernetesServerList(KubernetesClient client,
```

Comment:
Instead of adding the logic in this class itself for the service option, can we just add another `ServerList` bean and have the bean be enabled based on the configuration property?

Reply:
This is great, I can use an abstract class to pull out the public functions, and just switch mode create ServerList on configuration property

Reply:
I have added the relevant documentation